runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -229,63 +252,124 @@ end
 --- are saved to facilitate document synchronization in the response.
 ---
 ---@package
-function STHighlighter:send_request()
+---@param range? lsp.Range
+function STHighlighter:send_request(range)
   local version = util.buf_versions[self.bufnr]
 
   self:reset_timer()
 
   for client_id, state in pairs(self.client_state) do
     local client = vim.lsp.get_client_by_id(client_id)
+    if client then
+      local current_result = state.current_result
+      local active_requests = state.active_requests
+
+      -- Capability checks
```

Comment:
Since these capabilities support dynamic registration, we should use `client:supports_method` (which takes into account support added after the initialize request).